tokio-util/src/sync/mpsc.rs

```diff
@@ -64,13 +64,45 @@ async fn make_acquire_future<T>(
     }
 }
 
-impl<T: Send + 'static> PollSender<T> {
+#[derive(Debug)]
+// TODO: This should be replace with a type_alias_impl_trait to eliminate `'static` and all the transmutes
+struct PollSenderFuture<T>(ReusableBoxFuture<'static, Result<OwnedPermit<T>, PollSendError<T>>>);
+
+impl<T> From<ReusableBoxFuture<'static, Result<OwnedPermit<T>, PollSendError<T>>>>
+    for PollSenderFuture<T>
+{
+    fn from(v: ReusableBoxFuture<'static, Result<OwnedPermit<T>, PollSendError<T>>>) -> Self {
+        Self(v)
+    }
+}
+
+impl<T: Send> From<Option<Sender<T>>> for PollSenderFuture<T> {
+    fn from(data: Option<Sender<T>>) -> Self {
+        let v = ReusableBoxFuture::<'_, Result<OwnedPermit<T>, PollSendError<T>>>::new(
+            make_acquire_future(data),
+        );
+        // This is safe because we're just transmuting the `'static` bound on the Box<dyn> to get around storage issue
+        unsafe { mem::transmute(v) }
+    }
+}
+
+impl<T: Send> PollSenderFuture<T> {
+    fn get<'a>(&mut self) -> &mut ReusableBoxFuture<'a, Result<OwnedPermit<T>, PollSendError<T>>>
```

Comment:
Please replace this with a `set` and `poll` method instead. The `set` method should be unsafe because the caller promises that the future does not actually live past the lifetime. Callers of `set` will need to document why they ensure that. The `poll` method can be safe, and you don't need to transmute the lifetime to call it.

Reply:
Done, but I've modified `set` to not need to be unsafe by taking the `Sender` instead of just any `Future`